pkg/cloudprovider/aws/instance.go

```diff
@@ -70,15 +70,15 @@ func (p *InstanceProvider) Create(ctx context.Context,
 	if err != nil {
 		return nil, fmt.Errorf("creating fleet %w", err)
 	}
-	// TODO aggregate errors
-	if count := len(createFleetOutput.Errors); count > 0 {
-		return nil, fmt.Errorf("errors while creating fleet, %v", createFleetOutput.Errors)
-	}
 	if count := len(createFleetOutput.Instances); count != 1 {
-		return nil, fmt.Errorf("expected 1 instance, but got %d", count)
+		return nil, fmt.Errorf("expected 1 instance, but got %d due to errors %v", count, createFleetOutput.Errors)
```

Comment:
What does this error message look like?

Reply:
["Your requested instance type (c5a.4xlarge) is not supported in your requested Availability Zone (us-east-2a). Please retry your request by not specifying an Availability Zone or choosing us-east-2b, us-east-2c.", 
.... ]